sphinx/ext/intersphinx/__init__.py

```diff
@@ -0,0 +1,83 @@
+"""Insert links to objects documented in remote Sphinx documentation.
+
+This works as follows:
+
+* Each Sphinx HTML build creates a file named "objects.inv" that contains a
+  mapping from object names to URIs relative to the HTML set's root.
+
+* Projects using the Intersphinx extension can specify links to such mapping
+  files in the `intersphinx_mapping` config value.  The mapping will then be
+  used to resolve otherwise missing references to objects into links to the
+  other documentation.
+
+* By default, the mapping file is assumed to be at the same location as the
+  rest of the documentation; however, the location of the mapping file can
+  also be specified individually, e.g. if the docs should be buildable
+  without Internet access.
+"""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+import sphinx
+
+from ._cli import inspect_main
+from ._load import (
+    fetch_inventory,
+    fetch_inventory_group,
+    load_mappings,
+    normalize_intersphinx_mapping,
+)
+from ._resolve import (
+    IntersphinxDispatcher,
+    IntersphinxRole,
+    IntersphinxRoleResolver,
+    install_dispatcher,
+    inventory_exists,
+    missing_reference,
+    resolve_reference_any_inventory,
+    resolve_reference_detect_inventory,
+    resolve_reference_in_inventory,
+)
+from ._shared import InventoryAdapter
+
+if TYPE_CHECKING:
+    from sphinx.application import Sphinx
+    from sphinx.util.typing import ExtensionMetadata
+
+
+__all__ = (
```

Comment:
Do we put `__all__` before or after imports ? I personally put it at the very top so that when you open a file, you directly see what is being exported but I'm not sure what convention we should take.

Reply:
I'd like to see us follow PEP8 guidance on this.

However: doing that would be best achieved with linting support.  I've filed a feature request #12179 for that.. but it may require retaining `flake8`.

Reply:
I recommend using PEP 8.

Reply:
moved to before imports